frame/staking/src/mock.rs

```diff
@@ -505,6 +505,10 @@ pub fn active_era() -> EraIndex {
 	Staking::active_era().unwrap().index
 }
 
+pub fn current_era() -> EraIndex {
```

Comment:
given your description: 

> The active era should only be used for rewards stuff.

I think you can now remove all occurrences of acrive_era() and remove the function up there ^^.